### ISSUE TITLE
auto-compile being skipped every time

### ISSUE THREAD

**Issue by gusdleon** [xezno/sbox-quixel-bridge#8]
I'm having a issue importing assets since about a month, getting the error  
`The Asset system didn't catch asset quickly enough - skipping auto-compile for asset '{quixelAsset.Name}'`

And getting a warning in the error list of sandbox (and visual studio) about the async task `ImportFrom` not having an await operator

I solved both problems inserting a `await Task.Delay(500);` below the line 44 of this file
https://github.com/xezno/sbox-quixel-bridge/blob/6533172e783ef004a8062be321fd69e761ce2516/code/BridgeImporter.cs#L44-L54

but didn't created a PR before because I don't feel that's the correct approach to that problem.

what do you think?

**Comment by xezno** [comment]
Yeah the s&box asset system propagates changes really slowly.. I don't think there's any events either for it unfortunately. It might be worth me asking for some and then just hooking this up to those.

Not having auto-compile isn't a huge deal right now so it's not worth hacking around.. assets will compile whenever they're needed, this is just a convenience thing for people importing lots of assets at once

**Comment by xezno** [comment]
Just found out "assetsystem.changes" exists while I was doing something completely unrelated, it's undocumented for whatever reason, I'll have a look into using that

**Comment by sabahmax-inc** [comment]
Hi! Thanks for digging into this — I opened PR #12 to fix #8. It adds a small delay in `BridgeImporter.cs` so the asset system has time to pick up the asset before auto-compile runs, which should stop the “didn’t catch asset quickly enough” error and the related async warning.

**Comment by gusdleon** [comment]
sus
